src/app.py

```diff
@@ -29,21 +29,23 @@ def initialize(self):
 
     def setup_players(self):
         tokens = self.cli.get_player_tokens()
-        builder = PlayerBuilder(self.validator, self.processor, self.cli)
-        players = builder.build_players(tokens)
-
+        choice = self.cli.get_opponent_choice()
+        self.check_choice_for_exit(choice)
+        builder = PlayerBuilder(self.processor, self.cli)
+        players = builder.build_players(tokens, choice)
         self.players = players
 
     def setup_board(self):
-        board_type = self.cli.get_board_type()
-        builder = BoardBuilder()
-        board = builder.build_board(board_type)
-        self.board = board
+        board_choice = self.cli.get_board_type()
+        self.check_choice_for_exit(board_choice)
+        chosen_board = int(board_choice) - 1
+        board_type = constants.BOARD_CHOICES[chosen_board]
+        self.board = Board(board_type)
 
-    def setup_validator(self):
-        builder = ValidatorBuilder()
-        self.validator = builder.build_validator(self.processor)
+    def check_choice_for_exit(self, choice):
```

Comment:
The name is a bit deceiving. Check infers that it doesn't actually do anything. However, you actually exit if the check passes.